tower-sessions-core/src/session.rs

```diff
@@ -371,16 +371,26 @@ impl Session {
     /// // Data is cleared...
     /// assert!(session.get::<usize>("foo").await.unwrap().is_none());
     ///
+    /// // ...data is cleared before loading from the backend...
+    /// let session = Session::new(session.id(), store.clone(), None);
+    /// session.clear().await;
+    /// assert!(session.get::<usize>("foo").await.unwrap().is_none());
+    ///
     /// let session = Session::new(session.id(), store, None);
-    /// // ...but not deleted from the store.
+    /// // ...but data is not deleted from the store.
```

Comment:
Oh, I had missed it but I find this comment a bt confusing, should probably add "unless saved explicitly or through the service layer"?
(Because we set modified so ultimately in axum usage it is deleted from the store)

Reply:
I can see why this is a bit confusing. What will happen is the service will save the blank record to the store. But it won't delete the record from the store. That's an important distinction because it means the session can be reused by future requests whereas that is not the case with flush.